RFS/src/main/java/com/rfs/common/LuceneDocumentsReader.java

```diff
@@ -17,58 +21,116 @@
 @RequiredArgsConstructor
 @Slf4j
 public class LuceneDocumentsReader {
+    public static Function<Path, LuceneDocumentsReader> getFactory(boolean softDeletesPossible, String softDeletesField) {
+        return path -> new LuceneDocumentsReader(path, softDeletesPossible, softDeletesField);
+    }
     public static final int NUM_DOCUMENTS_BUFFERED = 1024;
+
     protected final Path indexDirectoryPath;
+    protected final boolean softDeletesPossible;
+    protected final String softDeletesField;
+
+    /**
+     * There are a variety of states the documents in our Lucene Index can be in; this method extracts those documents
+     * that would be considered "live" from the ElasticSearch/OpenSearch perspective.
+     *
+     * For context, when ElasticSearch/OpenSearch deletes a document, it doesn't actually remove it from the Lucene Index.
+     * Instead, what happens is that the document is marked as "deleted" in the Lucene Index, but it is still present in the
+     * Lucene segment on disk.  The next time that segment is merged, the deleted documents are removed from the Lucene Index.
```

Comment:
Technically, the documents are not propagated into the new segment.  Documents are never removed from a segment.  From a nomenclature perspective (I'm always fuzzy on what's an index and a segment), wouldn't the document have been not-present both before and after merge in the index?  I guess the question is, does the index include the tombstones or not?

Having a link to a high-level explanation of the lucene storage structure could be helpful.